src/components/viewer/report-editor/ReportEditor.tsx

```diff
@@ -0,0 +1,113 @@
+  AnnotationReportResourceType, AnnotationResourceType,
+  Reasons,
+} from 'api/annotations';
+import { hideEditor } from '../../../store/actions';
+import Report from './Report';
+import { PPScopeClass } from '../../../class_consts';
+import Suggestion from './Suggestion';
+
+interface IReportEditorProps {
+  annotation: AnnotationAPIModel;
+  onCancel: (e) => void;
+  onSuccess: () => void;
+  createAnnotationReport: (instance: AnnotationReportAPICreateModel) => Promise<AnnotationReportAPIModel>;
+}
+
+interface IReportEditorState {
+  activeDialog: 'menu'|'report'|'suggestion';
+}
+
+enum Dialogs {
+  MENU = 'menu',
+  REPORT = 'report',
+  SUGGESTION = 'suggestion',
+}
+
+@connect(
+  state => state,
+  dispatch => ({
+    createAnnotationReport: (instance: AnnotationReportAPICreateModel) => {
+        return dispatch(createResource(instance));
+    },
+  }),
+)
+export default class ReportEditor extends React.Component<Partial<IReportEditorProps>, Partial<IReportEditorState>> {
+
+  static defaultState = {
+    activeDialog: Dialogs.MENU,
+  };
+
+  constructor(props: IReportEditorProps) {
+    super(props);
+    this.state = ReportEditor.defaultState;
+  }
+
+  selectDialog = (e) => {
+    // We use e.currentTarget (the event handling element) since in Chrome e.target returns the node inside button
+    this.setState({ activeDialog: e.currentTarget.value });
+  }
+
+  save = (reason: Reasons, comment: string) => {
+    const instance = {
+      type: AnnotationReportResourceType,
+      attributes: { reason, comment },
+      relationships: {
+        annotation: {
+          data: {
+            id: this.props.annotation.id,
+            type: AnnotationResourceType,
+          },
+        },
+      },
+    };
+    this.props.createAnnotationReport(instance).then(() => {
+        this.props.onSuccess();
+      })
+      .catch((errors) => {
+        console.log(errors);
+      });
+  }
+
+  render() {
+    const {
+      annotation,
+      onCancel,
+    } = this.props;
+
+    switch (this.state.activeDialog) {
+      case 'menu':
+        return (
+          <div className={classNames(PPScopeClass, styles.self, styles.menu)}>
+            <div>
+              <button onClick={this.selectDialog} value={Dialogs.REPORT}>
+                <span className={classNames(styles.reportIcon)} />
+                <span> Zgłoś przypis </span>
+              </button>
+            </div>
+            <div>
+              <button onClick={this.selectDialog} value={Dialogs.SUGGESTION}>
+                <span className={classNames(styles.suggestIcon)} />
+                <span> Zasugeruj poprawkę </span>
+              </button>
+            </div>
+          </div>
+        );
+      case 'report':
+        return <Report annotation={annotation} onCancel={onCancel} onSubmit={this.save}/>;
+      case 'suggestion':
+        return <Suggestion annotation={annotation} onCancel={onCancel} onSubmit={this.save}/>;
+      default:
```

Comment:
I like the naming distinction between `onSubmit` and `onSuccess` based on whether the data is dispatched to the server and there is room for failure -- I think it's a nice distinction to always keep.